### PR TITLE
Update the dockerfile to be Debian based, not Alpine based

### PR DESCRIPTION
**Pull request by hawkowl**
This means we don't use libressl instead of openssl, and we get to use a lot of the manylinux1 wheels vs building them with headers ourselves.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -55,24 +49,19 @@ RUN pip install --prefix="/install" --no-warn-script-location \
 ### Stage 1: runtime
 ###
 
-FROM docker.io/python:${PYTHON_VERSION}-alpine3.10
+FROM docker.io/python:${PYTHON_VERSION}-slim
 
-# xmlsec is required for saml support
-RUN apk add --no-cache --virtual .runtime_deps \
-        libffi \
```
Comment (richvdh) [inline docker/Dockerfile]: we need a bunch of this stuff for features that people expect to work in the docker image.

Comment (hawkowl) [reply docker/Dockerfile]: so because we use the available wheels, we don't, because cffi and etc ship what's required.

**Comment by hawkowl** [comment]
I added some `--version` functionality so that we can check that Synapse, as well as its optional dependencies, are functional. On build, I get:

```
Synapse/1.6.0rc1

Additional functionality:
matrix-synapse-ldap3   [  OK  ]
postgres               [  OK  ]
resources.consent      [  OK  ]
acme                   [  OK  ]
saml2                  [  OK  ]
systemd                [NOT OK]
url_preview            [  OK  ]
sentry                 [  OK  ]
opentracing            [  OK  ]
jwt                    [  OK  ]
```

File: docker/Dockerfile (hunk 1 above)
Comment (richvdh) [reply docker/Dockerfile]: I think we still need xmlsec (or, as debian calls it, `xmlsec1`) at least. pysaml relies on the `/usr/bin/xmlsec1` binary.

**Comment by richvdh** [comment]
for future reference: this is probably a thing we should do if only to bring the docker image into line with other distros wrt which ssl library we use (see https://cryptography.io/en/latest/installation/)

**Comment by richvdh** [comment]
Having said that: can't we just build against `openssl-dev` rather than `libressl-dev`? (see https://pkgs.alpinelinux.org/contents?file=libssl.so&path=&name=&branch=edge&arch=x86_64)

**Comment by richvdh** [comment]
related: #7000

**Comment by clokep** [comment]
I'm going to close this since it hasn't been updated in ~6 months. If this is necessary in the future we can re-open this PR.

**Pull request closed without merging**